### ISSUE TITLE
Tokens inside of macro call and attr

### ISSUE THREAD

**Issue by dtolnay** [rust-analyzer/ungrammar#7]
TokenTree is currently defined with the rule `'(' ')' | '{' '}' | '[' ']'`. I would expect the rule to allow tokens within those delimiters, as well as other kinds of tokens like identifiers and literals and punctuation.

https://github.com/rust-analyzer/ungrammar/blob/07d5806c9345e43c5fb2baa427bc149c8808a10f/rust.ungram#L60-L66

https://github.com/rust-analyzer/ungrammar/blob/07d5806c9345e43c5fb2baa427bc149c8808a10f/rust.ungram#L286-L287

**Comment by matklad** [comment]
Yeah, this is somewhat of a placeholder node at the moment. We can list every token here, but that doesn’t seems useful for the purpose of genearating a syntax tree, and probably will be confusing with respect to contextual tokens & jointness.

Definitelly needs a comment in the source thoug.

**Comment by mjbshaw** [comment]
Instead of trying to list every possible token, could just a generic `'token'*` be used? That is:

```
 TokenTree = 
   '(' 'token'* ')' 
 | '{' 'token'* '}' 
 | '[' 'token'* ']' 
```

**Comment by Veykril** [comment]
Closing as the grammar file is no longer maintained
